### ISSUE TITLE
Can't view or edit data

### ISSUE THREAD

**Issue by MerielP** [ox-it/ords#717]
To reproduce, go to the database page for an ORDS database on Dev, and click 'View, Edit, and Query Data'. This produces an error message saying that the page doesn't exist.

**Comment by thestoat** [comment]
Test

**Comment by thestoat** [comment]
I shall add a "consistency" column in the databases table in the project page that will check the database data is consistent (as far as ORDS can check)

**Comment by thestoat** [comment]
My guess is that the data no longer exists (this is a dev system, after all, and data can be deleted without much notice ... though admittedly that shouldn't happen). I am currently waiting for logging to be enabled so that I can work on the specific problem Meriel is seeing (it doesn't happen locally for me). In the mean time, as mentioned above, I shall add a data consistency column: though this sort of problem should never happen in a live environment, the possibility that it might has become a lot worse since some users are now able to get at their data via ODBC, and thus compromise data integrity without ORDS' knowledge.
I think this will take 2 days for a very simplified version of this work. To be completely accurate about data integrity, ORDS would need to check what it thinks is the correct schema using its own tables and then match that precisely against what currently exists. This will take more time to implement and, of course, may render the system too slow to use in practise. For now some simple checks will bed made.

**Comment by thestoat** [comment]
Please check now and report if the problem still exists. Hopefully this particular problem has now been fixed.

**Comment by jajwilson** [comment]
Alas there are still no tables being listed in projects on Dev, nor is there any database structure to view - just an empty sheet of grid paper. Interestingly I _can_ view the saved queries for each database. Looking at the databases on the project screen I can see that all of my databases in the project are listed either as "Good (0 tables found for main)" or "Problem Detected" under the 'Data Integrity' column. The number of database versions for each database is being reported accurately as either 1 or 2.

**Comment by jajwilson** [comment]
Alas it’s not sorted yet – did you get updated by GitHub when I added my comment just now, or do I need to notify to this kind of thing manually?

James

**Comment by thestoat** [comment]
This should be fixed now. If you are still having troubles accessing existing databases then try uploading a new one and report back.

**Comment by MerielP** [comment]
Sorry - still having problems. The problems are also getting increasingly weird.

I tried viewing my existing databases, and found that I still couldn't see any of the tables. The 'Data integrity' column for most databases said 'Good (0 tables found for main)'. In a few cases, it said 'Problem detected' - nowhere was the table count above zero.

As suggested, I therefore tried uploading a new database. The upload process seemed to go smoothly - except that even while the upload process was still happening, ORDS was telling me I had two versions of the database, which was clearly false.

After I got the email notification that my upload had been successful, I went to try to view the data. No luck, sadly - I'm still getting a blank table list, and ORDS is once again reporting the data integrity as 'Good (0 tables found for main)'.

But... what's this? I now find that all the _other_ databases in the same project now have what looks like the right table count! The process of uploading a new database seems to have resulted in ORDS spotting that these databases do, in fact, contain data.

When I go to the table list, it looks promising - there are links to all my tables, just as there should be. However, clicking on a table name does not actually open the table, but simply takes me back to the project page. So, sadly, I still can't view or edit data.

**Comment by thestoat** [comment]
Currently you won't be able to see existing tables with the new code. This is because permissions are set up when you upload the tables, and your tables were set up before this code came along. I need to look at how this will work on App. My temptation is to manually set the correct permissions on App once the new code is promoted since it is a one off operation - but if there are too many and the work will therefore take too long then I may need to write a tool to do this. That means, of course, that you won't be able top get at any of your old data on dev. If that is a problem then let me know.

However, you should be able to upload and use a database. Please send me a sample that doesn't work so I can see this problem for myself.

**Comment by MerielP** [comment]
Hurrah! I deleted the database I uploaded earlier and uploaded it a second time, and this time it seems to be working - I can view the data and edit it.

If it is possible for access to my data on Dev to be restored, that would be good. I'd got various sample databases set up there for use in the documentation (including one guide that I'm halfway through writing). I could probably reproduce these if needs be, but it would be a pain.
